### PR TITLE
Add release compilation mode and output name options

### PR DESCRIPTION
**Pull request by teggotic**
Fixes https://github.com/jemc/mare/issues/46

### CODE REVIEW

File: src/mare/compiler.cr [hunk 1]
```diff
@@ -1,4 +1,19 @@
 module Mare::Compiler
+  class CompilerOptions
+    property release
+    property no_debug
+    property binary_name
+
+    DEFAULT_BINARY_NAME = "main.out"
```
Comment (jemc) [inline src/mare/compiler.cr:7]: Why `main.out` instead of `main`?

File: src/mare/compiler/code_gen/debug_info.cr [hunk 2]
```diff
@@ -11,6 +11,7 @@ class Mare::Compiler::CodeGen
       @mod : LLVM::Module,
       @builder : LLVM::Builder,
       @target_data : LLVM::TargetData,
+      @generate_debug_info = true,
```
Comment (jemc) [inline src/mare/compiler/code_gen/debug_info.cr:14]: Is this variable used anywhere, or is it replaced by `ctx.options.no_debug`?
```suggestion
```

File: src/mare/compiler/binary_verona.cr [hunk 3]
```diff
@@ -37,18 +37,31 @@ class Mare::Compiler::BinaryVerona
...
+        "-O0"
+      end
+
     link_args << obj_filename
     link_args << VERONA_STATIC_LIB
-    link_args << "-o" << "main" # TODO: customizable output binary filename
+    link_args << "-o" << ctx.options.binary_name
 
     res = Process.run("/usr/bin/env", link_args, output: STDOUT, error: STDERR)
+    # res = Process.run("/usr/bin/env", ["strip", ctx.options.binary_name], output: STDOUT, error: STDERR)
```
Comment (jemc) [inline src/mare/compiler/binary_verona.cr:57]: Looks like this line can be removed, right?
```suggestion
```

File: src/mare/compiler.cr (hunk 1 above)
Comment (teggotic) [reply src/mare/compiler.cr:7]: I thought that main is less common name coz we will overwrite this file.

File: src/mare/compiler/binary_verona.cr (hunk 3 above)
Comment (teggotic) [reply src/mare/compiler/binary_verona.cr:57]: Forgot to remove it

File: src/mare/compiler/code_gen/debug_info.cr (hunk 2 above)
Comment (teggotic) [reply src/mare/compiler/code_gen/debug_info.cr]: Forgot to remove it

File: src/mare/compiler/binary_verona.cr (hunk 3 above)
Comment (teggotic) [reply src/mare/compiler/binary_verona.cr:57]: I just was thinking if we need to strip our files

Comment (teggotic) [reply src/mare/compiler/binary_verona.cr:57]: We can actually strip files in release mode to reduce size of the executable

Comment (jemc) [reply src/mare/compiler/binary_verona.cr:57]: I'm okay if you want to put the `strip` command in - I just want don't want to leave a commented-out line in. So I'd prefer to either remove it or uncomment it.

**Pull request merged**